esp-hal-embassy/src/executor/thread.rs

```diff
@@ -2,28 +2,27 @@
 use core::marker::PhantomData;
 
 use embassy_executor::{raw, Spawner};
-
-use crate::{
+use esp_hal_common::{
     atomic::{AtomicBool, Ordering},
     get_core,
     prelude::interrupt,
 };
-#[cfg(multi_core)]
-use crate::{
+#[cfg(any(feature = "esp32", feature = "esp32s3"))]
+use esp_hal_common::{
     interrupt,
     peripherals::{self, SYSTEM},
 };
 
 /// global atomic used to keep track of whether there is work to do since sev()
 /// is not available on either Xtensa or RISC-V
-#[cfg(not(multi_core))]
+#[cfg(not(any(feature = "esp32", feature = "esp32s3")))]
```

Comment:
Completely off-topic but maybe we need a `esp-metadata` crate which contains the tomls etc from the soc module. This would allow crates other than esp-hal to use these nice cfgs!